### PR TITLE
chore|style: code-style validation through checkstyle and reformat according to tractus-x style guide

### LINKED ISSUE #57: Use proper code style/formatting
The source code of this project does not follow any code style guide. 

- The indentation is done partly with spaces, partly with tabs.
- Partly there are spaces at the end of lines
- Partly there are empty lines with some spaces
- Partly [a blank line at the end of files](https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline) is missing.
- Sometimes spaces are set around the equal sign in variable assignments, sometimes not.
- Sometimes the code is way too nested and not decoupled enough. E.g. I discovered a [function](https://github.com/eclipse-tractusx/knowledge-agents-edc/blob/64e0598b84f14e4979040e0741732138d238e4f3/agent-plane/agent-plane-protocol/src/main/java/org/eclipse/tractusx/agents/edc/AgreementController.java#L123-L163) which goes 8 levels deep
- Sometimes if-statements use a space between `if` and the `(`, sometimes not.
- ...

## Suggestion

Pick a code style, follow it and enforce it with some automated tests in the CI pipeline.

For general files (not java) I also recommend to use a proper IDE/Editor which trims automatically trailing spaces, etc.
Also I would recommend to add an [`.editorconfig` file](https://editorconfig.org/). Most of the IDE's/Editors are supporting it out of the box.

### LINKED ISSUE #76: Maven Plugin: Update Eclipse Dash Tool in pom.xml
In your `pom.xml` you're referencing a quite old dash tool version:

https://github.com/eclipse-tractusx/knowledge-agents-edc/blob/17b88aa3da87a2891723410f3214e8789eaaba98/pom.xml#L205-L214

Consider using a more up to date version, e.g. `1.0.2-SNAPSHOT` or `1.0.3-SNAPSHOT`.

### PR DESCRIPTION
**Pull request by drcgjung**
## WHAT

This PR introduces checkstyle into the build process based on the Tractus-X style guide for Java.
It also performs the whole reformatting necessary on the source files.
Finally it did a DEPENDENCY check using a more up-to-date dash version.

## WHY

Developers from other parts of Tractus-X were confused by the current approach (which was mostly checked with spotless and without any specific rules).

## FURTHER NOTES

Closes #57 <-- _insert Issue number if one exists_
Closes #76 

May conflict with other PRs of higher priority in which case this needs to be resolved.

### CODE REVIEW

File: agent-plane/agent-plane-protocol/src/main/java/org/eclipse/tractusx/agents/edc/SkillStore.java [hunk 1]
```diff
@@ -22,10 +22,11 @@
 /**
  * interface to a skill store
  */
-public interface ISkillStore {
```
Comment (Martin0815bla) [inline agent-plane/agent-plane-protocol/src/main/java/org/eclipse/tractusx/agents/edc/SkillStore.java:25]: What is the convention for Interfaces in KA? Not to have it with a leading capital I ?

File: agent-plane/agent-plane-protocol/src/main/java/org/eclipse/tractusx/agents/edc/service/DataspaceSynchronizer.java [hunk 2]
```diff
@@ -19,28 +19,34 @@
...
 import org.eclipse.tractusx.agents.edc.model.DcatCatalog;
 import org.eclipse.tractusx.agents.edc.model.DcatDataset;
-import org.eclipse.tractusx.agents.edc.rdf.RDFStore;
-import org.apache.jena.graph.Node;
-import org.apache.jena.graph.NodeFactory;
-import org.apache.jena.sparql.core.Quad;
-import org.eclipse.edc.spi.monitor.Monitor;
-import org.eclipse.edc.spi.query.Criterion;
-import org.eclipse.edc.spi.query.QuerySpec;
+import org.eclipse.tractusx.agents.edc.rdf.RdfStore;
```
Comment (Martin0815bla) [inline agent-plane/agent-plane-protocol/src/main/java/org/eclipse/tractusx/agents/edc/service/DataspaceSynchronizer.java:40]: What is the purpose of renaming RDFStore to RdfStore when e.g. org.apache.jena.riot.system.StreamRDFLib has still an upper case RDF in it?

File: agent-plane/agent-plane-protocol/src/main/java/org/eclipse/tractusx/agents/edc/SkillStore.java (hunk 1 above)
Comment (drcgjung) [reply agent-plane/agent-plane-protocol/src/main/java/org/eclipse/tractusx/agents/edc/SkillStore.java:25]: Relevant is the tractus-x convention/style guide. A leading I is no more allowed, instead the interface has a normal name and the implementaiton could be named SkillStoreImpl or so.

File: agent-plane/agent-plane-protocol/src/main/java/org/eclipse/tractusx/agents/edc/service/DataspaceSynchronizer.java (hunk 2 above)
Comment (drcgjung) [reply agent-plane/agent-plane-protocol/src/main/java/org/eclipse/tractusx/agents/edc/service/DataspaceSynchronizer.java:40]: checkstyle only checks tractus-x code ... apache jena can do whatever they want. And in tractus-x, multiple capital letters are not allowed, we should use Camel case.

File: agent-plane/agent-plane-protocol/src/main/java/org/eclipse/tractusx/agents/edc/service/DataspaceSynchronizer.java [hunk 3]
```diff
@@ -53,26 +59,26 @@ public class DataspaceSynchronizer implements Runnable {
     /**
      * constants
      */
-    protected final static Node CX_ASSET=NodeFactory.createURI("https://w3id.org/catenax/ontology/common#offers");
-    protected final static Map<String,Node> assetPropertyMap=new HashMap<>();
-    protected final static QuerySpec federatedAssetQuery = QuerySpec.Builder.newInstance().
-            filter(List.of(new Criterion("https://w3id.org/catenax/ontology/common#isFederated","=","true^^xsd:boolean"))).build();
+    protected static final Node CX_ASSET = NodeFactory.createURI("https://w3id.org/catenax/ontology/common#offers");
+    protected static final Map<String, Node> ASSET_PROPERTY_MAP = new HashMap<>();
```
[the remark this answers is not in the document]
Comment (drcgjung) [reply agent-plane/agent-plane-protocol/src/main/java/org/eclipse/tractusx/agents/edc/service/DataspaceSynchronizer.java:63]: static finals are constants. To clearly distinguish constants from dynamic fields they are written in upper case (non-camel-case)

**Review by Martin0815bla — APPROVED** [review]

**Comment by drcgjung** [comment]
@carslen with my best regards, here is another checkstyle introduction that has been already checked by @Martin0815bla for possible IP changes
I merged it with the recent fixes from main (which needed some reformatting).

**Review by SebastianBezold — APPROVED** [review]
While this is a huge PR, that would require a project contribution IP check, the changes are almost without exception formatting ones. 

So frine from my point of view. With `.editorconfig` in place this should not happen too often 😁

**Pull request merged**